src/azure-openapi-validator/rule.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { JsonPath } from "../jsonrpc/types";
+
+export enum OpenApiTypes {
+  "default" = 0,
+  "arm" = 1 << 0,
+  "DataPlane" = 1 << 1
+}
+
+export enum MergeStates {
+  "individual",
+  "composed"
+}
+
+interface ValidationMessage {
+  message: string;
+  location: JsonPath;
+}
+
+export interface Rule {
+  readonly id: string; // see Mxxx codes on https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md
```

Comment:
I took a look at https://github.com/Azure/azure-rest-api-specs/blob/9b50c0eed8704ad714a21b5a9cc876b00dbf9d11/documentation/openapi-authoring-automated-guidelines.md
Looks like we are still using ```Mxxx``` while the rules have been updated to ```S/Rxxx```
Where do we add new ids for rules. @kirthik